src/librustc_resolve/diagnostics.rs

```diff
@@ -702,13 +763,13 @@ impl<'a> Resolver<'a> {
                         name_binding.is_extern_crate() && lookup_ident.span.rust_2018();
 
                     let is_visible_to_user =
-                        !in_module_is_extern || name_binding.vis == ty::Visibility::Public;
+                        parent_or_parent_sibling || name_binding.vis == ty::Visibility::Public;
```

Comment:
I suspect that you are trying to hand-roll `is_accessible_from` here.

Could you try using it instead of collecting the parent module set manually?

Reply:
Yes, managed to use `is_ancestor_of`, and it seems to work as expected.

Reply:
I mean `parent_or_parent_sibling` can also be avoided.

Reply:
Indeed it can be avoided. It also fixed another case relating to `crate_visibility_modifier`. This is neat!